x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/advanced_detector_modal/advanced_detector_modal.tsx

```diff
@@ -427,17 +424,8 @@ function createDefaultDescription(dtr: RichDetector) {
   return detectorToString(basicDetector);
 }
 
-// fixes issue with EuiComboBox.
-// if the options list only contains one option and nothing has been selected, set
-// selectedOptions list to be an empty array
-function createSelectedOptions(
-  selectedOption: EuiComboBoxOptionOption,
-  options: EuiComboBoxOptionOption[]
-): EuiComboBoxOptionOption[] {
-  return (options.length === 1 && options[0].label !== selectedOption.label) ||
-    selectedOption.label === ''
-    ? []
-    : [selectedOption];
+function createSelectedOptions(selectedOption: EuiComboBoxOptionOption): EuiComboBoxOptionOption[] {
+  return selectedOption === undefined || selectedOption.label === '' ? [] : [selectedOption];
```

Comment:
it appears the original `EuiComboBox` fix is no longer needed. There is no adverse behaviour if the list only contains one item and nothing has been selected.